### ISSUE TITLE
Data access request seems to go to file contributor instead of dataset contact

### ISSUE THREAD

**Issue by tosunmerve** [IQSS/dataverse#9976]
**What steps does it take to reproduce the issue?**

Requesting access to restricted files where the file uploader does not correspond with dataset contact.

* When does this issue occur?

Requesting access to restricted files where the file uploader does not correspond with dataset contact.

* Which page(s) does it occurs on?

Dataset-page, under Files.

* What happens?

When requesting acces to restricted files, the access request seems to automatically end up with the file uploader instead of the appointed dataset contact.

* To whom does it occur (all users, curators, superusers)?

All users.

* What did you expect to happen?

We expect that all access requests go to the dataset manager(s) (i.e. dataset contact), and not individual file uploaders.

**Which version of Dataverse are you using?**

5.12

**Any related open or closed issues to this bug report?**

No.

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*

**Comment by pdurbin** [comment]
@tosunmerve hi! Thanks for reporting an issue.

Just to make sure I understand, user1 is requesting access to a file. You expect an email/notification to go to user2 who is the dataset contact. Instead the email/notification goes to user3 who uploaded the file. Is that right?

Judging from the code below, the notification is going to any person who has `ManageDatasetPermissions` on the dataset, not just the person who uploaded the file.

```
permissionService.getUsersWithPermissionOn(Permission.ManageDatasetPermissions, dataset).stream().forEach((au) -> {
    userNotificationService.sendNotification(au, ts, UserNotification.Type.REQUESTFILEACCESS, fileId, null, requestor, true);
});
```

The thing about the dataset contact is that they don't even necessarily have an account in the system. It's just an email address and maybe a name. It's part of the metadata for a dataset.

**Comment by tosunmerve** [comment]
Dear Philip,

Thank you for clarifying. Yes, this is indeed the issue. If I understand you correctly, contacts are not defined as persons, but metadata? Considering this, is it still possible to send the notification to the contact, or does the current function not allow that?

If it is possible, can you show us how to adapt the code you shared with us?
If it is not possible currently, would it be possible to describe contacts as persons in the next update of Dataverse?

Warm regards,

Merve

From: Philip Durbin ***@***.***>
Reply to: IQSS/dataverse ***@***.***>
Date: Thursday, 5 October 2023 at 20:24
To: IQSS/dataverse ***@***.***>
Cc: Merve Tosun ***@***.***>, Mention ***@***.***>
Subject: Re: [IQSS/dataverse] Data access request seems to go to file contributor instead of dataset contact (Issue #9976)


@tosunmerve<https://github.com/tosunmerve> hi! Thanks for reporting an issue.

Just to make sure I understand, user1 is requesting access to a file. You expect an email/notification to go to user2 who is the dataset contact. Instead the email/notification goes to user3 who uploaded the file. Is that right?

Judging from the code below, the notification is going to any person who has ManageDatasetPermissions on the dataset, not just the person who uploaded the file.

permissionService.getUsersWithPermissionOn(Permission.ManageDatasetPermissions, dataset).stream().forEach((au) -> {

    userNotificationService.sendNotification(au, ts, UserNotification.Type.REQUESTFILEACCESS, fileId, null, requestor, true);

});

The thing about the dataset contact is that they don't even necessarily have an account in the system. It's just an email address and maybe a name. It's part of the metadata for a dataset.

**Comment by pdurbin** [comment]
Yes! Exactly! That's a good way to think about it. Dataset contacts are metadata not users.

In the code we refer to users as "authenticated users" which is why you see `au` in the code I mentioned. Here it is again:

```
userNotificationService.sendNotification(au...
```

If you want to make a pull request, it would probably easier to create a variation on that method that accepts an email address (a string) rather than trying to turn dataset contacts into authenticated users.

If you go down this route, I'd also suggest making this feature configurable: off by default but a feature installations can turn on.

I hope this makes sense! Please keep the questions coming!

**Comment by tosunmerve** [comment]
Dear Philip,

So if we adjust the code as we did below, the problem should be solved?

permissionService.getUsersWithPermissionOn(Permission.ManageDatasetPermissions, dataset).stream().forEach((au) -> {
    ***@***.***”, ts, UserNotification.Type.REQUESTFILEACCESS, fileId, null, requestor, true);
});

Thanks and best,

Merve

From: Philip Durbin ***@***.***>
Reply to: IQSS/dataverse ***@***.***>
Date: Thursday, 26 October 2023 at 20:46
To: IQSS/dataverse ***@***.***>
Cc: Merve Tosun ***@***.***>, Mention ***@***.***>
Subject: Re: [IQSS/dataverse] Data access request seems to go to file contributor instead of dataset contact (Issue #9976)


Yes! Exactly! That's a good way to think about it. Dataset contacts are metadata not users.

In the code we refer to users as "authenticated users" which is why you see au in the code I mentioned. Here it is again:

userNotificationService.sendNotification(au...

If you want to make a pull request, it would probably easier to create a variation on that method that accepts an email address (a string) rather than trying to turn dataset contacts into authenticated users.

If you go down this route, I'd also suggest making this feature configurable: off by default but a feature installations can turn on.

I hope this makes sense! Please keep the questions coming!

**Comment by pdurbin** [comment]
Something like that, yes. I think.
